packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -121,20 +149,62 @@ contract FeeHandler is
   // Without this the contract cant receive Celo as native transfer
   function() external payable {}
 
-  /**
-    @dev Sets the fee beneficiary address to the specified address.
-    @param beneficiary The address to set as the fee beneficiary.
-  */
-  function setFeeBeneficiary(address beneficiary) external onlyOwner {
-    return _setFeeBeneficiary(beneficiary);
+  function setCarbonFraction(uint256 newFraction) external onlyOwner {
+    _setCarbonFraction(newFraction);
+  }
+
+  function changeOtherBeneficiaryAllocation(
+    address beneficiary,
+    uint256 _newFraction
+  ) external onlyOwner {
+    FixidityLib.Fraction memory newFraction = FixidityLib.wrap(_newFraction);
+    _setBeneficiaryFraction(beneficiary, newFraction);
+  }
+
+  function addOtherBeneficiary(
+    address beneficiary,
+    uint256 _newFraction,
+    string calldata name
+  ) external onlyOwner {
+    require(otherBeneficiaries[beneficiary].exists == false, "Beneficiary already exists");
+    FixidityLib.Fraction memory newFraction = FixidityLib.wrap(_newFraction);
+
+    otherBeneficiaries[beneficiary].exists = true;
+    _setBeneficiaryFraction(beneficiary, newFraction);
```

Comment:
Nit: In both _setBeneficiaryFraction and _setBeneficiaryName we are reading from same storage again - we could just pass storage beneficiary into these functions

Reply:
wont work because of the emits